src/ArduinoIoTCloud.h

```diff
@@ -145,7 +147,9 @@ class ArduinoIoTCloudClass
 
     ConnectionHandler * _connection = nullptr;
     PropertyContainer _property_container;
-    TimeService _time_service;
+    TimeService * _time_service = ArduinoIoTCloudTimeService();
```

Comment:
Is there any specific reason why this is now a pointer? It looks like you want to employ the `Singleton` design pattern. Why? Usually (except for very few use cases) `Singleton`s are to be avoided.

Also please do all initialisation within the constructor, or all initialisation within the class body, but not mixing between those two options.

Reply:
I see. Still one small improvement would be to operate with references `&` instead of pointers `*`, then you never need to worry that a pointer can be zero, i.e.
```C++
TimeService & ArduinoIoTCloudTimeService() {
  static TimeService _timeService_instance;
  return _timeService_instance;
}
```

Reply:
Related: Why do you have `ArduinoIoTCloudTimeService` twice in code (Once in production code, once in test code). Would it not be sufficient to have it defined only once? I do not see MCU-specific code in that statement that would not compile for host.

Reply:
No MCU-specific code, but adding TimeService.cpp to the test system will require some extra work because then we will need also to stub/fake a bit on NTPUtils and maybe also RTC. So at the end i've decided to duplicate the function to make the build system happy and run the tests.

The only property that is actually using TimeService is CloudSchedule so i don't think adding the whole class would be really beneficial for the test coverage since CloudSchedule is implementing its own getTime() fake function in test_CluodSchedule.cpp file.

Reply:
> operate with references & instead of pointers *

please see https://github.com/arduino-libraries/ArduinoIoTCloud/pull/284/commits/920e920a7f4125e6272b8c6ba814301eb9f0442f